specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/webapplicationfirewall.json

```diff
@@ -948,6 +955,75 @@
         }
       }
     },
+    "ExceptionEntry": {
+      "required": [
+        "matchVariable",
+        "valueMatchOperator"
+      ],
+      "description": "Adds exception to allow a request when the condition is satisfied.",
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "enum": [
+            "RequestURI",
+            "RemoteAddr",
+            "RequestHeader"
+          ],
+          "description": "The variable on which we evaluate the exception condition",
+          "x-ms-enum": {
+            "name": "ExceptionEntryMatchVariable",
+            "modelAsString": true
+          }
+        },
+        "values": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Allowed values for the matchVariable"
+        },
+        "valueMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith",
+            "IPMatch"
+          ],
+          "x-ms-enum": {
+            "name": "ExceptionEntryValueMatchOperator",
+            "modelAsString": true
+          },
+          "description": "Operates on the allowed values for the matchVariable"
+        },
+        "selectorMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith"
+          ],
+          "description": "When the matchVariable points to a key-value pair (e.g, RequestHeader), this operates on the selector",
+          "x-ms-enum": {
+            "name": "ExceptionEntrySelectorMatchOperator",
+            "modelAsString": true
+          }
+        },
+        "selector": {
+          "type": "string",
+          "description": "When the matchVariable points to a key-value pair (e.g, RequestHeader), this identifies the key."
+        },
+        "exceptionManagedRuleSets": {
```

Comment:
> it doesn't make sense to let the user define the parts such as target is IP address, if the rule set isn't an IP address set of rules at all, but something else based on headers.

Let me add some context of exception and WAF managed rulesets:
Managed rulesets have multiple rules that evaluate different parts of the request, they are categorized into rule groups based on the kind of attack they can evaluate within which each rule can check for different parts of the request.

Exceptions have been a big customer ask to prevent false positives. What Cx want is a way to stop some rules from getting triggered when the request has certain IP/URL/header etc.

So it may very well happen that the customer requirement is to stop a rule that evaluates a header from getting triggered when the request is sent by a certain IP that they trust. That is exactly the kind of use case that brought the need for this feature.